inc/osvr/PluginKit/SkeletonInterfaceC.h

```diff
@@ -0,0 +1,123 @@
+/** @file
+    @brief Header
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef INCLUDED_SkeletonInterfaceC_h_GUID_69382E69_DFA1_4FB0_5287_D874B740B4C0
+#define INCLUDED_SkeletonInterfaceC_h_GUID_69382E69_DFA1_4FB0_5287_D874B740B4C0
+
+/* Internal Includes */
+#include <osvr/PluginKit/DeviceInterfaceC.h>
+#include <osvr/Util/ChannelCountC.h>
+#include <osvr/Util/ClientReportTypesC.h>
+
+// Library/third-party includes
+// - none
+
+// Standard includes
+// - none
+
+OSVR_EXTERN_C_BEGIN
+
+/** @defgroup PluginKitCSkeleton Skeleton interface (base C API)
+    @brief Sends notifications that skeleton reports (aka tracker reports) are
+   complete. Skeleton interface is implemented as a device that exposes a
+   tracker interface for each skeleton element. For each element, a separate
+   tracker report will be sent with an updated pose followed by
+   skeletonComplete. and the client can request a complete report of entire
+   skeleton (all reported elements), as well as subsets. Note that since it
+   exposes a device via multiple interfaces, you will need to "configure" both
+   interfaces (Tracker and Skeleton).
+    @ingroup PluginKit
+    @{
+*/
+
+/** @brief Opaque type used in conjunction with a device token to send data on
+    Skeleton Interface
+*/
+typedef struct OSVR_SkeletonDeviceInterfaceObject *OSVR_SkeletonDeviceInterface;
+
+/** @brief Specify that your device will implement the Skeleton interface.
+
+    @param opts The device init options object.
+    @param [out] iface An interface object you should retain with the same
+    lifetime as the device token in order to send messages conforming to a
+   Skeleton interface.
+   @param jsonDescriptor A device descriptor json that contains skeleton spec
+    @param numSensors The number of non-connected skeletons you will be
+   reporting. Note that the number of skeleton sensors is not the same as number
+   of skeleton elements. For example if plugin device reports two hands that
+   will be 2 separate skeleton sensors because hands are not connected via
+   common parent. If device can report an entire skeleton then it should only
+   report one sensor.
+*/
```

Comment:
It's not very clear what it means to be a "parent" here. So far, my understanding is that a skeleton is a collection of joints, with static relationships between some of them (knee connected to the hip, and so forth). A particular skeleton interface might support some or all of a complete skeleton. Should it matter in terms of number of sensors whether the parts supported by the plugin are connected physically? Maybe an analysis plugin will simulate the missing joints with physics/IK, but that's difficult to do in a generic way if some are single-sensor and some are multi-sensor.